version: '3.8'

services:
  cassandra:
    image: cassandra:3.11
    container_name: cassandra_db
    ports:
      - "9042:9042"
    volumes:
      - cassandra_data:/var/lib/cassandra
    environment:
      - CASSANDRA_START_RPC=true
    healthcheck:
      test: ["CMD-SHELL", "cqlsh -e 'describe keyspaces'"]
      interval: 30s
      timeout: 10s
      retries: 5
    networks:
      - cassandra-net

  python-app:
    build:
      context: ./Parte2
      dockerfile: Dockerfile
    container_name: python_app
    depends_on:
      cassandra:
        condition: service_healthy
    volumes:
      - ./Parte2:/app
      - ./Parte1:/2da convocatoria/Parte1 
    networks:
      - cassandra-net
    environment:
      - CASSANDRA_CONTACT_POINTS=cassandra
    stdin_open: true  
    tty: true        
    command: ["python", "main.py"]

volumes:
  cassandra_data:

networks:
  cassandra-net:
    driver: bridge
